### PR TITLE
Report uuid of test nodes to ScyllaDB

### PR DESCRIPTION
**Pull request by kongove**
We uses uuid to identify the Scylla db nodes, this patchset support to report uuid of all test nodes to ScyllaDB server, it helps to know the real users.

### CODE REVIEW

**Comment by slivne** [comment]
we need the report to be done when adding a node not only when booting up the cluster - can we do it in the node level to make sure we always have it.

**Comment by kongove** [comment]
The uuid is generated in setup (scylla_setup script) stage, we can only report after the setup.

**Comment by slivne** [comment]
I know that - the issue is that you added a call to the
report_housekeeping_uuid()
only at one point in tester.py


there are additional places that create a cluster / add nodes I am not sure
they all pass through this point.


for example the grow_cluster test adds a node - will the added node report
its uuid as scylla

**Comment by kongove** [comment]
The other places has their own setUp(), they can't pass through my report_housekeeping_uuid(). It's a problem.

Then I will add report action at the end of node_setup() of all backends.

**Comment by roydahan** [comment]
@kongove, I didn't see another commit for this PR to solve the issue @slivne raised.
Can you please push it so we can merge this?

**Comment by kongove** [comment]
@slivne @roydahan I just moved the reporting to the end of wait_for_up(). We call it in all node_setup() of all backends, all subtests will go through this point.

**Comment by roydahan** [comment]
@kongove, have you tested it with 1.6?
Please do and check it with: select * from housekeeping.marked;

**Comment by kongove** [comment]
@roydahan I just tested the pathset with 1.6, it can report uuid successfully. There is some new improvement:
1. server will record the IP to calling the method, so let's execute `curl` command in db nodes (not in sct test machine)
2. add a mark file to avoid repeatedly reporting.

The latest patchset has been tested, and it's safe to merge.

File: sdcm/cluster.py [hunk 1]
```diff
@@ -806,12 +806,28 @@ def wait_jmx_down(self, verbose=True):
...
 
+    def _report_housekeeping_uuid(self, verbose=True):
+        """
+        report uuid of test db nodes to ScyllaDB
+        """
+        uuid_path = '/etc/scylla.d/housekeeping.uuid'
+        report_status_path = '/etc/scylla.d/housekeeping.uuid.reported'
+        cmd = 'curl "https://i6a5h9l1kl.execute-api.us-east-1.amazonaws.com/prod/check_version?uu=%s&mark=scylla"'
+
+        if self.file_exists(uuid_path) and not self.file_exists(
```
Comment (lmr) [inline sdcm/cluster.py]: The check has to be done in the remote system, therefore using `self.remoter.run('test -e %s' % report_status_path)`

Comment (kongove) [inline sdcm/cluster.py]: Thanks for the catching.
`ignore_status=True` is also needed.

**Comment by kongove** [comment]
New update: 
v4: fix checking node files (Lucas), update files' path
The PR is tested.

File: sdcm/cluster.py [hunk 2]
```diff
@@ -806,12 +806,31 @@ def wait_jmx_down(self, verbose=True):
...
+
+        uuid_result = self.remoter.run('test -e %s' % uuid_path,
+                                       ignore_status=True, verbose=verbose)
+        mark_result = self.remoter.run('test -e %s' % mark_path,
+                                       ignore_status=True, verbose=verbose)
+        if uuid_result.exit_status == 0 and mark_result.exit_status != 0:
+            result = self.remoter.run('cat %s' % uuid_path, verbose=verbose)
+            self.remoter.run(cmd % result.stdout.strip())
+            self.remoter.run('sudo -u scylla touch %s' % mark_path,
+                             verbose=verbose)
```
Comment (lmr) [inline sdcm/cluster.py]: Now we're talking!

**Pull request merged**